ShopApp/ShopApp/ViewController.swift

```diff
@@ -7,14 +7,64 @@
 
 import UIKit
 import FirebaseDatabase
+import FirebaseAuth
+import FirebaseCore
+import GoogleSignIn
 
 class ViewController: UIViewController {
 
+    @IBOutlet private var signInButton: GIDSignInButton!
+
+    @IBAction private func googleSignIn() {
+        guard let clientID = FirebaseApp.app()?.options.clientID else { return }
+
+        // Create Google Sign In configuration object.
+        let config = GIDConfiguration(clientID: clientID)
+
+        // Start the sign in flow!
+        GIDSignIn.sharedInstance.signIn(with: config, presenting: self) { [unowned self] user, error in
+
+            if error != nil {
+                return
+            }
+
+            guard
+                let authentication = user?.authentication,
+                let idToken = authentication.idToken
+            else {
+                return
+            }
+
+            let credential = GoogleAuthProvider.credential(withIDToken: idToken, accessToken: authentication.accessToken)
+
+            Auth.auth().signIn(with: credential) { _, error in
+
+                if let error = error {
+                    _ = error as NSError
+                    print("Error")
+                } else {
+            // User is signed in
+                    print("signed")
```

Comment:
Extract these 5 lines of code in a separate method called for example "NavigateToHomeScreen" and call it here and in ViewDidLoad, since it's duplicated code.